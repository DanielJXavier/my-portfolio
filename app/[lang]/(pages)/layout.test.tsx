import "tests/matchMedia.mock";

import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import { useParams, usePathname } from "next/navigation";

import Layout from "./layout";

jest.mock("next/navigation");

(useParams as jest.Mock).mockImplementation(() => ({ lang: "en" }));

(usePathname as jest.Mock).mockImplementation(() => "/en");

describe("Pages layout", () => {
  it("renders the layout", () => {
    const { container } = render(<Layout>children</Layout>);

    expect(container).toMatchSnapshot();
  });
});
