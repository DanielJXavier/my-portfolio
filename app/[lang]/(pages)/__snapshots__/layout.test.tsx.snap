// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages layout renders the layout 1`] = `
<div>
  <header
    class="py-2 md:py-3 lg:py-4 shadow-sm shadow-primary"
    data-testid="header"
  >
    <div
      class="relative container mx-auto px-4 md:px-5 xl:px-6 grid grid-cols-[auto_1fr_auto] items-center justify-between justify-items-end gap-x-2 md:gap-x-3 lg:gap-x-8"
    >
      <a
        class="text-xl lg:text-3xl font-light hover:text-secondary text-secondary cursor-default"
        data-testid="header-link"
        href="/en"
      >
        &lt;DanielXavier /&gt;
      </a>
      <nav
        class="order-last md:order-none"
        data-testid="menu"
      >
        <button
          aria-label="Menu"
          class="w-8 h-[26px] p-1.5 flex items-center justify-center gap-y-1 flex-col [&_hr]:hover:border-t-secondary  md:hidden"
          data-testid="menu-button"
        >
          <hr
            class="w-5 border-t-2 border-t-primary "
          />
          <hr
            class="w-5 border-t-2 border-t-primary "
          />
          <hr
            class="w-5 border-t-2 border-t-primary "
          />
        </button>
        <ul
          class="absolute md:relative top-[38px] md:top-0 right-0 left-0 h-[calc(100vh-46px)] md:h-auto px-4 md:px-0 py-2 md:py-0 bg-background md:bg-transparent z-50 text-lg lg:text-xl font-light hidden md:flex flex flex-col md:flex-row gap-y-2 md:gap-x-3"
          data-testid="menu-list"
        >
          <li>
            <a
              class=" decoration-0 underline-offset-2 tracking-tighter [word-spacing:-0.25em] hover:text-secondary hover:underline"
              href="/en/experience"
            >
              Experience
            </a>
          </li>
          <li>
            <a
              class=" decoration-0 underline-offset-2 tracking-tighter [word-spacing:-0.25em] hover:text-secondary hover:underline"
              href="/en/education"
            >
              Education
            </a>
          </li>
          <li>
            <a
              class=" decoration-0 underline-offset-2 tracking-tighter [word-spacing:-0.25em] hover:text-secondary hover:underline"
              href="/en/skills"
            >
              Skills
            </a>
          </li>
          <li>
            <a
              class=" decoration-0 underline-offset-2 tracking-tighter [word-spacing:-0.25em] hover:text-secondary hover:underline"
              href="/en/recommendations"
            >
              Recommendations
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="w-7 lg:w-8 h-7 lg:h-8"
      />
    </div>
  </header>
  <main
    class="main-container"
  >
    children
  </main>
  <footer
    class="self-end pb-6 md:pb-8 xl:pb-12 grid gap-y-2 items-center justify-center content-center"
    data-testid="footer"
  >
    <menu
      class="flex gap-x-3 justify-center"
      data-testid="lang-switcher"
    >
      <li>
        <a
          class="pointer-events-none cursor-default"
          href="/en"
        >
          ðŸ‡ºðŸ‡¸
        </a>
      </li>
      <li>
        <a
          class=""
          href="/pt"
        >
          ðŸ‡§ðŸ‡·
        </a>
      </li>
    </menu>
    <p
      class="text-xs md:text-sm text-center"
    >
      Daniel
       
      Xavier
       Â© 
      2025
      .
       
      <br
        class="sm:hidden"
      />
      All rights reserved.
    </p>
  </footer>
</div>
`;
